lockup-factory/src/lib.rs

```diff
@@ -77,87 +75,60 @@ impl Default for LockupFactory {
 #[near_bindgen]
 impl LockupFactory {
     #[init]
-    pub fn new(master_account_id: AccountId,
-               lockup_master_account_id: AccountId,
-               whitelist_account_id: AccountId,
-               foundation_account_id: AccountId) ->
-               Self {
+    pub fn new(
+        master_account_id: ValidAccountId,
+        lockup_master_account_id: ValidAccountId,
```

Comment:
Actually this should be this contract account ID (`env::current_account_id()`) because you can't create a sub-account of another account. So just remove this argument. You may also want to check that the current account ID is shorted than the 23 characters, to make sure that `40` char lockup (from hash) plus `.` separator fits into the 64 account length limit 
```suggestion
```